sdk/communication/Azure.Communication.CallingServer/src/ContentDownloader.cs

```diff
@@ -0,0 +1,149 @@
+        {
+            HttpRange? pageRange = null;
+
+            if (range != default || startOffset != 0)
+            {
+                pageRange = new HttpRange(
+                    range.Offset + startOffset,
+                    range.Length.HasValue ?
+                        range.Length.Value - startOffset :
+                        null);
+            }
+
+            HttpMessage message = GetHttpMessage(sourceEndpoint, pageRange);
+
+            await _client._pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
+            switch (message.Response.Status)
+            {
+                case 200:
+                case 206:
+                    {
+                        Stream value = message.ExtractResponseContent();
+                        return Response.FromValue(value, message.Response);
+                    }
+                default:
+                    throw await _client._clientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
+            }
+        }
+
+        private Response<Stream> StartDownload(Uri sourceEndpoint, HttpRange range = default, long startOffset = 0, CancellationToken cancellationToken = default)
+        {
+            HttpRange? pageRange = null;
+
+            if (range != default || startOffset != 0)
+            {
+                pageRange = new HttpRange(
+                    range.Offset + startOffset,
+                    range.Length.HasValue ?
+                        range.Length.Value - startOffset :
+                        null);
+            }
+
+            HttpMessage message = GetHttpMessage(sourceEndpoint, pageRange);
+            _client._pipeline.Send(message, cancellationToken);
+
+            switch (message.Response.Status)
+            {
+                case 200:
+                case 206:
+                    {
+                        Stream value = message.ExtractResponseContent();
+                        return Response.FromValue(value, message.Response);
+                    }
+                default:
+                    throw _client._clientDiagnostics.CreateRequestFailedException(message.Response);
+            }
+        }
+
+        private HttpMessage GetHttpMessage(Uri sourceEndpoint, HttpRange? rangeHeader = null)
+        {
+            HttpMessage message = _client._pipeline.CreateMessage();
+            Request request = message.Request;
+            request.Method = RequestMethod.Get;
+            var uri = new RequestUriBuilder();
+            uri.Reset(sourceEndpoint);
+
+            request.Uri = uri;
+
+            if (rangeHeader != null)
+            {
+                request.Headers.Add(Constants.HeaderNames.Range, rangeHeader.ToString());
+            }
+
+            // Even if using an external location, we must use the acs resource's information to sign our request.
+            string path = uri.Path;
+            if (path.StartsWith("/", StringComparison.InvariantCulture))
```

Comment:
when is this true?

Reply:
Technically always, for instance (from my tests):
- Given `uri = {https://localhost/documents/metadata/document_id/acsmetadata}`
- then when doing `string path = uri.Path`, path contains `"/documents/metadata/document_id/acsmetadata"`

I decided to leave it as `if` in case something changes, i.e., maybe `Path` implementation, maybe the URL that AMS service returns, etc.